src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -1248,4 +1265,79 @@ export class WorkspacesUsecases {
   }> {
     return this.teamRepository.getTeamUserAndTeamByTeamId(userUuid, teamId);
   }
+
+  async getAvatarUrl(avatarKey: Workspace['avatar']): Promise<string> {
+    return await this.avatarService.getDownloadUrl(avatarKey);
+  }
+
+  async upsertAvatar(
+    workspaceId: Workspace['id'],
+    avatarKey: string,
+  ): Promise<Error | { avatar: string }> {
+    const workspace = await this.workspaceRepository.findOne({
+      id: workspaceId,
+    });
+
+    if (!workspace) {
+      throw new BadRequestException('Not valid workspace');
+    }
+
+    if (!avatarKey) {
+      throw new BadRequestException('Avatar key required');
+    }
+
+    if (workspace.avatar) {
+      try {
+        await this.avatarService.deleteAvatar(workspace.avatar);
+      } catch (err) {
+        Logger.error(
+          `[WORKSPACE/SET_AVATAR]Deleting the avatar in workspaceId: ${
+            workspace.id
+          } has failed. Error: ${(err as Error).message}`,
+        );
+        throw new Error('Deleting avatar to workspace has failed');
```

Comment:
Just propagate the error please. So we can catch/log later easily when we add a proper filterException. Nestjs will just show an internal server error unless you specify a specific HTTP error, so not worry about this error being parsed to the response.